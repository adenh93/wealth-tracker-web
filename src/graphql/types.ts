export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type CryptoCurrency = {
  __typename?: 'CryptoCurrency';
  id: Scalars['Int'];
  name: Scalars['String'];
  rank: Scalars['Int'];
  symbol: Scalars['String'];
};

export type CryptoCurrencyHolding = {
  __typename?: 'CryptoCurrencyHolding';
  cryptoCurrency?: Maybe<CryptoCurrency>;
  cryptoCurrencyId: Scalars['Int'];
  holdings: Scalars['Float'];
  id: Scalars['Int'];
  netHoldingsValue: Scalars['Float'];
  percentChange24h: Scalars['Float'];
  price: Scalars['Float'];
};

export type CryptoCurrencyHoldingsAddInput = {
  cryptoCurrencyId: Scalars['Int'];
  holdings: Scalars['Float'];
};

export type CryptoCurrencyHoldingsUpdateInput = {
  holdings: Scalars['Float'];
  id: Scalars['Int'];
};

export type Mutation = {
  __typename?: 'Mutation';
  cryptoCurrencyHoldingsAdd?: Maybe<CryptoCurrencyHolding>;
  cryptoCurrencyHoldingsDelete?: Maybe<CryptoCurrencyHolding>;
  cryptoCurrencyHoldingsUpdate?: Maybe<CryptoCurrencyHolding>;
};


export type MutationCryptoCurrencyHoldingsAddArgs = {
  input: CryptoCurrencyHoldingsAddInput;
};


export type MutationCryptoCurrencyHoldingsDeleteArgs = {
  id: Scalars['Int'];
};


export type MutationCryptoCurrencyHoldingsUpdateArgs = {
  input: CryptoCurrencyHoldingsUpdateInput;
};

export type Query = {
  __typename?: 'Query';
  cryptoCurrencies?: Maybe<Array<CryptoCurrency>>;
  cryptoCurrencyHoldings?: Maybe<Array<CryptoCurrencyHolding>>;
};


export type QueryCryptoCurrenciesArgs = {
  query?: Maybe<Scalars['String']>;
};
