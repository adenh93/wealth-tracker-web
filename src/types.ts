export interface Asset {
  id: string
  logoSrc: string
  name: string
  ticker: string
  price: number
  gain24Hr: number
  holdings: number
}
