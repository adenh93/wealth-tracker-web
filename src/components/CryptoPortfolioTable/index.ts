export { default } from './CryptoPortfolioTable'
export type { CryptoPortfolioTableProps } from './CryptoPortfolioTable'
