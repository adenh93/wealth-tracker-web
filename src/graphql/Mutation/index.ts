export { default as CRYPTOCURRENCY_HOLDINGS_ADD } from './cryptoCurrencyHoldingsAdd'
export { default as CRYPTOCURRENCY_HOLDINGS_UPDATE } from './cryptoCurrencyHoldingsUpdate'
export { default as CRYPTOCURRENCY_HOLDINGS_DELETE } from './cryptoCurrencyHoldingsDelete'
