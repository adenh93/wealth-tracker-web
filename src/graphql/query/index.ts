export { default as CRYPTO_CURRENCIES } from './cryptoCurrencies'
