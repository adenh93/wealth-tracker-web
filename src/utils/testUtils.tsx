import { ReactElement } from 'react'
import { ThemeProvider } from '@mui/material'
import { render as tlRender, RenderResult } from '@testing-library/react'
import theme from '../theme'

export const render = (component: ReactElement): RenderResult =>
  tlRender(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
