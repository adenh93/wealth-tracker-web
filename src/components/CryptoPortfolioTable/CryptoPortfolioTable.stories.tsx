import { Story } from '@storybook/react'
import { CryptoCurrencyHolding } from '../../graphql/types'
import PortfolioTable, {
  CryptoPortfolioTableProps,
} from './CryptoPortfolioTable'

const config = {
  title: 'Components/PortfolioTable',
  component: PortfolioTable,
}

const Template: Story<CryptoPortfolioTableProps> = (args) => (
  <PortfolioTable {...args} />
)

export const Primary = Template.bind({})

const assets: CryptoCurrencyHolding[] = [
  {
    cryptoCurrency: {
      id: 1,
      rank: 1,
      name: 'Bitcoin',
      symbol: 'BTC',
    },
    id: 2,
    cryptoCurrencyId: 1,
    holdings: 15.25,
    price: 80532.5,
    percentChange24h: 1.54,
    netHoldingsValue: 1228120.625,
  },
  {
    cryptoCurrency: {
      id: 1027,
      rank: 2,
      name: 'Ethereum',
      symbol: 'ETH',
    },
    id: 2,
    cryptoCurrencyId: 1027,
    holdings: 30,
    price: 4302.25,
    percentChange24h: 4.32,
    netHoldingsValue: 129067.5,
  },
  {
    cryptoCurrency: {
      id: 1839,
      rank: 3,
      name: 'Binance Coin',
      symbol: 'BNB',
    },
    id: 3,
    cryptoCurrencyId: 1839,
    holdings: 45,
    price: 678.11,
    percentChange24h: 2.76,
    netHoldingsValue: 30514.95,
  },
]

Primary.args = {
  assets,
}

export default config
