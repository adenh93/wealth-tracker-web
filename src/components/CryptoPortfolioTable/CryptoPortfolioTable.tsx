import { FC } from 'react'
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableRow,
  Paper,
} from '@mui/material'
import CryptoPortfolioAsset from '../CryptoPortfolioAsset'
import TableHeadText from '../typography/TableHeadText'
import { StyledTableHead } from './CryptoPortfolioTable.css'
import { CryptoCurrencyHolding } from '../../graphql/types'

export interface CryptoPortfolioTableProps {
  assets: CryptoCurrencyHolding[]
  loading: boolean
  onEdit: (asset: CryptoCurrencyHolding) => void
  onRemove: (asset: CryptoCurrencyHolding) => void
}

const CryptoPortfolioTable: FC<CryptoPortfolioTableProps> = ({
  assets,
  loading,
  onEdit,
  onRemove,
}) => (
  <TableContainer component={Paper}>
    <Table>
      <StyledTableHead>
        <TableRow>
          <TableCell align="left">
            <TableHeadText>Asset</TableHeadText>
          </TableCell>
          <TableCell align="left">
            <TableHeadText>Price</TableHeadText>
          </TableCell>
          <TableCell align="left">
            <TableHeadText>Holdings</TableHeadText>
          </TableCell>
          <TableCell />
        </TableRow>
      </StyledTableHead>
      <TableBody>
        {assets.map((asset: CryptoCurrencyHolding) => (
          <CryptoPortfolioAsset
            key={asset.id}
            asset={asset}
            onEdit={onEdit}
            onRemove={onRemove}
          />
        ))}
      </TableBody>
    </Table>
  </TableContainer>
)

export default CryptoPortfolioTable
