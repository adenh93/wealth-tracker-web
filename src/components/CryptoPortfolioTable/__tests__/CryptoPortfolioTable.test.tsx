import CryptoPortfolioTable from '..'
import { CryptoCurrencyHolding } from '../../../graphql/types'
import { render } from '../../../utils/testUtils'

describe('CryptoPortfolioTable component tests', () => {
  const mockAssets: CryptoCurrencyHolding[] = [
    {
      cryptoCurrency: {
        id: 1,
        rank: 1,
        name: 'Bitcoin',
        symbol: 'BTC',
      },
      id: 1,
      cryptoCurrencyId: 1,
      holdings: 15.25,
      price: 80532.5,
      percentChange24h: 1.54,
      netHoldingsValue: 1228120.625,
    },
    {
      cryptoCurrency: {
        id: 1027,
        rank: 2,
        name: 'Ethereum',
        symbol: 'ETH',
      },
      id: 2,
      cryptoCurrencyId: 1027,
      holdings: 30,
      price: 4302.25,
      percentChange24h: 4.32,
      netHoldingsValue: 129067.5,
    },
  ]

  test('matches snapshot', () => {
    const { asFragment } = render(
      <CryptoPortfolioTable assets={mockAssets} loading={false} />
    )

    expect(asFragment).toMatchSnapshot()
  })

  test('it renders correctly', () => {
    const { container, getByText } = render(
      <CryptoPortfolioTable assets={mockAssets} loading={false} />
    )

    const rows = container.querySelectorAll('tbody tr')

    expect(rows).toHaveLength(2)
    expect(getByText('Bitcoin')).not.toBeNull()
    expect(getByText('Ethereum')).not.toBeNull()
  })
})
