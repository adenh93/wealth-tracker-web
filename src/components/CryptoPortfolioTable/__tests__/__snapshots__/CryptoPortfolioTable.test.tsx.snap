// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoPortfolioTable component tests matches snapshot 1`] = `[Function]`;
